source/languages/en/riakcs/upgrade-v20.md

```diff
@@ -0,0 +1,215 @@
+   ...
+]}.
+```
+
+Now it is just as follows at Riak 2.0.
+
+```
+buckets.default.allow_mult = true
+```
+
+Note: this is defined in riak_kv.schema
+
+#### Backend configuration
+
+If [riak_kv/1082](https://github.com/basho/riak_kv/pull/1082) is
+merged as it is, new configuration will be
+
+```
+storage_backend = prefix_multi
+cs_version = 20000
+```
+
+Note: otherwise (or early Riak 2.0) just requires same storage backend
+configuration in `advanced.config` as required in CS 1.4 or 1.5.
+
+#### Storage calculation
+
+If
+[storage calculation](http://docs.basho.com/riakcs/latest/cookbooks/Usage-and-Billing-Data/#Storage-Statistics)
+is required to your system, several more configurations are required.
+
+Including CS path: add following sentence to `advanced.config`:
+
+```erlang
+{riak_kv, [
+  {add_paths, ["/usr/lib/riak-cs/lib/riak_cs-x.y.z/ebin"]}
+]}.
+```
+
+If multibag is to be used, add multibag ebin path:
+
+```erlang
+{riak_kv, [
+  {add_paths, ["/usr/lib/riak-cs/lib/riak_cs-x.y.z/ebin",
+               "/usr/lib/riak-cs/lib/riak_cs_multibag-a.b.c/ebin"]}
+]}.
+```
+
+### Riak CS 1.5 to 2.0, including Stanchion
+
+Consult
+[Configuration Mapping Table](https://github.com/basho/riak_cs/wiki/Configuration-Mapping-Table-between-1.5-and-2.0-%5BRFC%5D)
+to preserve same configuration between CS 1.5 and 2.0.
+
+If obsolete configurations like `fold_objects_for_list_keys`,
+`n_val_1_get_requests` or `gc_paginated_indexes` are still set as
+`false`, **removing any of them from configuraiton file** is strongly
+recommended at Riak CS 2.0, because those configuration items are to
+preserve old and slow behaviours of Riak CS and have no bad effect on
+performance.
+
+#### Multibag configurations
+
+Multibag configurations are to be moved to `advanced.config` for both
+Riak CS and Stanchion.
+
+## Notes on Upgrading from Riak CS < 1.5.4
+
+[Some objects changes their names](https://github.com/basho/riak_cs/blob/develop/RELEASE-NOTES.md#notes-on-upgrading)
+after upgrade. Applications may change it's behaviour due to this bugfix.
+
+## Notes on Upgrading from Riak CS < 1.5.1
+
+[Bucket number limitation per user](https://github.com/basho/riak_cs/blob/develop/RELEASE-NOTES.md#notes-on-upgrading-1)
+had been introduced at 1.5.1. Users who have more than 100 buckets
+cannot create any bucket after upgrade unless the limit number
+extended at the system configuration.
+
+## Special Notes on Upgrading From Riak CS 1.4
+
+An operational procedure
+[to clean up incomplete multipart under deleted buckets](https://github.com/basho/riak_cs/blob/develop/RELEASE-NOTES.md#notes-on-upgrading-2)
+is needed. Otherwise new buckets which used to exist in the past
+couldn't be created. The operation will fail with 409 Conflict.
+
+## Riak CS nodes not corriding in a same box with live Riak node
+
+Riak CS nodes not corriding in a same box with live Riak node can be
+upgraded at any time while corresponding remote Riak is alive.
```

Comment:
Are both CS2.0-to-Riak1.4 and CS1.5-to-Riak2.0?

I'm thinking about the configurtion where Riak CS connects to
(potentially many) Riak via L4 reverse proxy like HAProxy.